JesusOmarR/react-apprenticeship-capstone1
# Router challenge

## Description
```
I made changes for the last mini challenge
I created a private route and a login to authenticate the users and made private routes to keep the favorites page just for register users
I created a favorite page to display all the video located in the local storage as favorite videos
Also a page for the details of each video and a list with all the favorites videos

```
## Type of Change


## How should this be tested?
```
run npm run test -a to run all the tests cases and see if they have a good implementation
```

## What is the deliverable?
```
React apprenticeship mini challenge 5
```
## Screenshots and links
```
![image](https://user-images.githubusercontent.com/97203630/150076395-e3ff984d-5a49-4127-bc3d-216f1cabb7e0.png)

<img width="1301" alt="image" src="https://user-images.githubusercontent.com/97203630/150077089-9a2ea903-0f52-4ca8-a1b3-4bb2ef3c9347.png">

<img width="1218" alt="image" src="https://user-images.githubusercontent.com/97203630/150077171-719ee306-1c3d-4f17-a67f-c6bda77bb701.png">

```
## Notes
```
In the local host a mock service. worker is implemented so all the api call are going to be intercepted
for a better test use the netifly link or disable the service worker
```
## Checklist
- [x ] I added the necessary documentation, if appropriate.
- [x ] I added tests to prove that my fix is effective or my deliverable works.
- [ x] I reviewed existing Pull Requests before submitting mine.
